File: go/leap/leap.go
package leap

const TestVersion = 1

// IsLeapYear takes a year and returns true if said year is a leap year or
// false if it is not
func IsLeapYear(year int) bool {
	return year%400 == 0 && year%4 == 0 || year%4 == 0 && year%100 != 0
}

File: javascript/leap/leap.js
var Leap = function(year) {
	this.year = year
};

Leap.prototype.isLeap = function() {
	if (this.year % 4 != 0) {
		return false;
	}

	if (this.year % 100 == 0 && this.year % 400 != 0) {
		return false;
	}

	return true
}

module.exports = Leap